plugins/javaembed/HpccClassLoader.java

```diff
@@ -32,7 +34,13 @@ private HpccClassLoader(java.net.URL [] urls, ClassLoader parent, int _bytecodeL
     }
     public Class<?> findClass(String className) throws ClassNotFoundException
     {
-        return defineClassForEmbed(bytecodeLen, bytecode, className.replace(".","/"));
+        Class result = (Class) classes.get(className);
```

Comment:
does this need to be thead safe?

Reply:
I wondered the same thing, and concluded that it PROBABLY doesn't need to be at the moment as a classloader object is not shared by more than one function context. But that may change in the future and probably smart to put a synchronized on the method now.